kotlin-inject-compiler-core/src/main/kotlin/me/tatarka/inject/compiler/InjectGenerator.kt

```diff
@@ -232,4 +230,32 @@ fun AstMethod.isProvider(): Boolean =
     !hasAnnotation<Provides>() && isAbstract && when (this) {
         is AstFunction -> parameters.isEmpty()
         is AstProperty -> true
-    } && receiverParameterType == null && returnType.isNotUnit()
\ No newline at end of file
+    } && receiverParameterType == null && returnType.isNotUnit()
+
+private fun dumpGraph(astClass: AstClass, entries: List<MethodEntry>): String {
```

Comment:
Maybe move this into its own file? Seems like pretty generic logic

Reply:
Kinda meh about that as it's only used here.